qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,153 @@
+
+        if (os_name == os_path) and (plat_name == arch_path):
+            if driver_dir in need_test_driver:
+                return True
+
+    def match_cat_and_verified_file(filterd_cat_file):
+        """
+        Tuple each selected '.cat' file and with same directory '.sys' file;
+        Tuple each selected '.cat' file and with same directory '.inf' file;
+        Tuple each selected '.cat' file and with same directory 'Wdf' file.
+
+        Collect all tuples in virtio_drivers list to be used signtool test.
+
+        :param filterd_cat_file:each filterd suffix '.cat' file
+        """
+        pattern = r'\w:\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        match_ret = re.match(pattern, filterd_cat_file)
+        driver_dir = match_ret.group()
+
+        driver_sys = filter(lambda x: driver_dir in x, drivers['.sys'])
+        driver_inf = filter(lambda x: driver_dir in x, drivers['.inf'])
+        driver_wdf = filter(lambda x: driver_dir in x, drivers['Wdf'])
+
+        cat_sys = map(lambda x: (filterd_cat_file, x), driver_sys)
+        cat_inf = map(lambda x: (filterd_cat_file, x), driver_inf)
+        cat_wdf = map(lambda x: (filterd_cat_file, x), driver_wdf)
+        cat_parm = cat_sys + cat_inf + cat_wdf
+        virtio_drivers.extend(cat_parm)
+
     try:
-        for drive in drive_list.split():
-            for type in ['.cat', '.sys']:
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        key = "VolumeName like 'virtio-win%'"
+        disk_letter = utils_misc.get_win_disk_vol(session, condition=key) + ":"
+        drive_volume_list.append(disk_letter)
+
+        for drive in drive_volume_list:
+            for type in ['.cat', '.sys', '.inf', 'Wdf']:
                 driver = session.cmd_output(list_files_cmd % (drive, type)).\
                     splitlines()[1:-1]
                 drivers[type] = driver
-            files = zip(drivers['.cat'], drivers['.sys'])
-            for cat, sys in files:
-                cmd = signtool_cmd % (cat, sys)
-                s, result = session.cmd_status_output(cmd)
-                if s:
-                    msg = "Fail command: %s. Output: %s" % (cmd, result)
-                    raise error.TestFail(msg)
+
+            cat_list = filter(filter_all_tested_cat_files, drivers['.cat'])
+            map(match_cat_and_verified_file, cat_list)
+
+            for driver_file in virtio_drivers:
+                test_cmd = signtool_cmd % (driver_file[0], driver_file[1])
+                test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+                status, result = session.cmd_status_output(test_cmd)
+                if status:
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
+                    error_context.context("Fail command: %s. Output: %s" %
+                                          (test_cmd, result), logging.info)
                 results_all += result
```

Comment:
better to add \n after add each result to results_all

Reply:
lijin, with currently code, the result.log info like follows, there are 2 space lines between 2 results connection.
=================================================================
Successfully verified: A:\amd64\Win2008R2\netkvm.sys

Number of files successfully Verified: 1
Number of warnings: 0
Number of errors: 0


Verifying: A:\amd64\Win2008R2\netkvm.inf
File is signed in catalog: A:\amd64\Win2008R2\netkvm.cat
Hash of file (sha1): F6B3680CE9A1396DED7D5E09EBF2A8B2CBE5277A
=================================================================